functions/dir_clean.pp

```diff
@@ -0,0 +1,21 @@
+# @summary Take a path and normalises it to its unix form.
+# Instead of having to deal with the different seperators between unix and windows this
+# function instead formates windows paths a equivalent unix like path.
+#
+# AFAIK this type of unix like path will work on windows.
+#
+# @param dir The path to clean
+# @return Stdlib::Absolutepath The cleaned path
+# @example clean Unix paths to return `/tmp/test/libs` (i.e. noop)
+#   extlib::dir_clean('/tmp/test/libs')
+# @example Clean Windows paths to return `/c/test/libs`
+#   extlib::dir_clean('c:\\'test\\libs')
+#
+# $dir is defined as String as it also accepts 'c:' which is not a valid
+# Stdlib::Absolutepath
+function extlib::dir_clean(String $dir) >> Stdlib::Unixpath {
```

Comment:
This doesn't look like an entirely correct validation. `Stdlib::Unixpath` is an absolute path while `String` could be a relative path. I don't know if Puppet validations return types, but `extlib::dir_clean('hi')` on a Linux kernel would probably fail. Makes a good testcase.

Reply:
Have updated the signature to take `Variant[Stdlib::Absolutepath, Pattern[/\A[a-zA-Z]:\z/]]`